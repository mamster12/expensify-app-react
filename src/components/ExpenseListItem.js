import React from 'react';
import { Link } from 'react-router-dom';

const ExpenseListItem = ({ description, amount, createdAt, id }) => {
    return (
        <div>
            <Link to={`/edit/${id}`}>
                <h3>{description}</h3>
            </Link>
            <ul>
                <li>amount: {amount}</li>
                <li>createdAt: {createdAt}</li>
            </ul>
        </div>
    );
};

export default ExpenseListItem;
