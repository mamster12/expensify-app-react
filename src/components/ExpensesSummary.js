import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import getTotalExpenses from '../selectors/expenses-total';
import numeral from 'numeral';

export const ExpensesSummary = (props) => (
    <div>
        <h1>
            Viewing {props.expenseCount}{' '}
            {props.expenseCount > 1 ? 'expenses' : 'expense'} totalling{' '}
            {numeral(props.expensesTotal / 100).format('$0,0.00')}
        </h1>
    </div>
);

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);

    return {
        expenseCount: visibleExpenses.length,
        expensesTotal: getTotalExpenses(visibleExpenses)
    };
};

export default connect(mapStateToProps)(ExpensesSummary);
