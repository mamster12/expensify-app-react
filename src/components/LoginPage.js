import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export const LoginPage = ({ startLogin }) => {
    return (
        <div>
            This is a login Page
            <button onClick={startLogin}>Login</button>
        </div>
    );
};

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);
