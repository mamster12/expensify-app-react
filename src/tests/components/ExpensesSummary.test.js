import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('Should render single expense Summary correctly', () => {
    const wrapper = shallow(
        <ExpensesSummary expenseCount={1} expensesTotal={100} />
    );
    expect(wrapper).toMatchSnapshot();
});

test('Should render multiple expenses Summary correctly', () => {
    const wrapper = shallow(
        <ExpensesSummary expenseCount={2} expensesTotal={100} />
    );
    expect(wrapper).toMatchSnapshot();
});
