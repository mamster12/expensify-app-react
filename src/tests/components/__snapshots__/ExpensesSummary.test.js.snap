// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render multiple expenses Summary correctly 1`] = `
<div>
  <h1>
    Viewing 
    2
     
    expenses
     totalling
     
    $1.00
  </h1>
</div>
`;

exports[`Should render single expense Summary correctly 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling
     
    $1.00
  </h1>
</div>
`;
